pages/admin/[id].tsx

```diff
@@ -62,23 +66,96 @@ export const getServerSideProps: GetServerSideProps = async ({
     }
   });
   tournament = JSON.parse(JSON.stringify(tournament));
-  players = JSON.parse(JSON.stringify(players));
+  const playerList = JSON.parse(JSON.stringify(players));
   rings = JSON.parse(JSON.stringify(rings));
   return {
-    props: { tournament, players, rings }
+    props: { tournament, playerList, rings }
   };
 };
 
-export default function Tournament({ tournament, players, rings }) {
+export default function Tournament({ tournament, playerList, rings }) {
+  const [players, setPlayers] = useState(playerList);
+  const handlePlayerStatus = (playerState, id) => {
+    const data = { state: playerState };
+    fetch(`/api/player/${id}/state`, {
+      method: "PATCH",
+      body: JSON.stringify(data)
+    });
+    const playerToBeUpdated = players.find((p) => p.id == id);
+    const updatedPlayer = { ...playerToBeUpdated, state: playerState };
+    setPlayers(
+      players.map((player) => (player.id !== id ? player : updatedPlayer))
+    );
+  };
   return (
     <AuthenticationRequired>
       <div>
-        <h2>{tournament.name}</h2>
-        <TournamentRings
-          tournament={tournament}
-          players={players}
-          rings={rings}
-        />
+        <h2 style={{ width: "100%" }}>{tournament.name}</h2>
+        <Grid container>
+          <Grid item xs={12} md={6}>
+            <div style={{ paddingLeft: "10px" }}>
+              <h2>Pelaajat</h2>
+              <table>
+                <tbody>
+                  {players.map((player) => (
+                    <tr key={player.id}>
+                      <td>
+                        <Link href={`/tournaments/users/${player.user.id}`}>
+                          <a>
+                            {player.user.firstName} {player.user.lastName}
+                          </a>
+                        </Link>
+                      </td>
+                      {player.state == "ACTIVE" ? (
+                        <td>
+                          <button
+                            onClick={() =>
+                              handlePlayerStatus("DEAD", player.id)
+                            }
+                          >
+                            Tapa
+                          </button>
```

Comment:
Hitsi kun nääkin buttonit olis hauska refaktoroida tästä omiksi funktioikseen. Toi `handlePlayerStatus`-funkkari vaan lienee ongelma, kun se käsittelee tän elementin sisällä määriteltyä hookia... Mitenkähän noita hookeja käytettäis niin, että ne ei pakottais käytännössä kaikkea tilankäsittelykoodia inlineksi 🤔

Reply:
Okei, keinoja on useampiakin: https://stackoverflow.com/a/53455474 

Eri asia tosin, onko just yksittäinen button nyt varsinaisesti lopulta sellainen, joka kannattaisi ekstrahoida omaksi funkkarikseen.